script.lua

```diff
@@ -7733,16 +7733,93 @@ function onObjectDestroy(obj)
         end
     end
 end
-function getCardImageURL(card)
-    local cardData = card.getData()
-    if cardData.CustomDeck then
-        for _, data in pairs(cardData.CustomDeck) do
-            if data.NumWidth == 1 and data.NumHeight == 1 then
-                return data.FaceURL
+function getReminderLocation(params)
+    local obj = params.obj
+    if obj == nil then
+        return nil
+    end
+
+    -- Get it from the script state if it's there
+    local state = {}
+    if obj.script_state ~= "" then
+        state = JSON.decode(obj.script_state)
+    end
+    if state.reminder ~= nil then
+        return state.reminder
+    end
+
+    -- Otherwise, use a sensible default
+    if obj.type == "Card" then
+        return {
+            field = "FaceURL",
+            x = -0.17,
+            y = -0.45,
+            width = 1.85,
+            height = 2.59,
+        }
+    elseif obj.hasTag("Spirit") then
+        return {
+            field = "ImageSecondaryURL",
+            x = 0.63,
+            y = -0.23,
+            width = 2.40,
+            height = 1.60,
+        }
+    else
+        return nil
+    end
+end
+function getReminderImageAttributes(params)
+    local objSize = 200 -- The height/width of UI panel to be the same size as the object
+
+    local obj = params.obj
+    if obj == nil then
+        return {image = ""}
+    end
+
+    local location = params.location
+    if location == nil then
+        location = getReminderLocation({obj = obj})
+    end
+
+    local imageURL = ""
+    local data = obj.getData()
+    if obj.type == "Card" then
+        if data.CustomDeck then
+            local deckID, deckData = next(data.CustomDeck)
+            deckID = tonumber(deckID)
+            imageURL = deckData[location.field]
+            local row = 0 -- 0-indexed
+            local column = 0 -- 0-indexed
+            if data.CardID and deckID then
+                local cardIndex = data.CardID - 100 * deckID -- 0-indexed
+                if cardIndex >= 0 and cardIndex < deckData.NumWidth * deckData.NumHeight then
+                    column = cardIndex % deckData.NumWidth
+                    row = (cardIndex - column) / deckData.NumWidth
+                end
             end
+            location.x = location.x + ((deckData.NumWidth - 1) / 2 - column) * location.width
```

Comment:
0, 0 is in the bottom right corner right?

Reply:
For the grid of card images, card 0 is the top left, card 1 is the top row, one across from the left, and so on. I've converted that (`cardIndex`) to `row` and `column`, which are still 0-indexed counting from the top left.

Meanwhile, UI coordinates put (0,0) in the *bottom* left; this mismatch is why the calculation for the x coordinate is an addition while the calculation for the y coordinate is a subtraction.

Further complicating things is the fact that here, we're not placing the mask on top of the image, we're placing the image behind the mask. Thus, to show something further to the right in the image, we need to slide the image itself left, and so on. That's why both `column` and `row` are negated in the calculations for `x` and `y` respectively.

Reply:
ah maybe that's where things got backwards you're not moving the mask but the image. was mainly confused because it seemed like things were shifting the opposite way